packages/contract/erc20.ts

```diff
@@ -0,0 +1,91 @@
+import { Contract, Interface, parseUnits, toNumber } from "ethers"
+
+import type { ContractRunner } from "~packages/node"
+import number from "~packages/util/number"
+import type { BigNumberish, HexString } from "~typing"
+
+export class ERC20Contract {
+  public readonly address: HexString
+  public name: string
+  public symbol: string
+  public decimals: number
+
+  private token: Contract
+  private static abi = [
+    "function balanceOf(address account) external view returns (uint256)",
+    "function transfer(address to, uint256 value) public returns (bool)",
+    "function transferFrom(address from, address to, uint256 value) public returns (bool)",
+    "function name() public view returns (string)",
+    "function symbol() public view returns (string)",
+    "function decimals() public view returns (uint8)"
+  ]
+
+  public static async init(address: HexString, runner: ContractRunner) {
+    const token = new Contract(address, ERC20Contract.abi, runner)
+    return new ERC20Contract(
+      token,
+      address,
+      await token.name(),
+      await token.symbol(),
+      toNumber(await token.decimals())
```

Comment:
This class is about contract, name, symbol or decimals should not be intrinsic fields for it. I would suggest provide them as methods for client to query at need.